File: Python/program/Vehicle.py
# Saya Boy Aditya Rohmaulana NIM 2203488 mengerjakan
# soal Latihan Praktikum 4 dalam mata kuliah DPBO
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin

class Vehicle:
    __plat_nomor = ""
    __merk = ""
    __tahun_produksi = ""
    __warna = ""
    
    def __init__(self, plat_nomor = "", merk = "", tahun_produksi = "", warna = ""):
        self.__plat_nomor = plat_nomor
        self.__merk = merk
        self.__tahun_produksi = tahun_produksi
        self.__warna = warna

    def get_plat_nomor(self):
        return self.__plat_nomor

    def set_plat_nomor(self, plat_nomor):
        self.__plat_nomor = plat_nomor

    def get_merk(self):
        return self.__merk

    def set_merk(self, merk):
        self.__merk = merk

    def get_tahun_produksi(self):
        return self.__tahun_produksi

    def set_tahun_produksi(self, tahun_produksi):
        self.__tahun_produksi = tahun_produksi

    def get_warna(self):
        return self.__warna

    def set_warna(self, warna):
        self.__warna = warna

File: CPP/program/Vehicle.cpp
/*Saya Boy Aditya Rohmaulana NIM 2203488 mengerjakan
soal Latihan Praktikum 4 dalam mata kuliah DPBO
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/

#pragma once

#include <string>
using namespace std;

class Vehicle {
  private:
    string plat_nomor;
    string merk;
    int tahun_produksi;
    string warna;

  public:
    //   Constructor
    Vehicle() {
    }

    Vehicle(string plat_nomor) {
        this->plat_nomor = plat_nomor;
        merk = "";
        tahun_produksi = -1;
        warna = "";
    }

    Vehicle(string plat_nomor, string merk) {
        this->plat_nomor = plat_nomor;
        this->merk = merk;
        tahun_produksi = -1;
        warna = "";
    }

    Vehicle(string plat_nomor, string merk, int tahun_produksi) {
        this->plat_nomor = plat_nomor;
        this->merk = merk;
        this->tahun_produksi = tahun_produksi;
        warna = "";
    }

    Vehicle(string plat_nomor, string merk, int tahun_produksi, string warna) {
        this->plat_nomor = plat_nomor;
        this->merk = merk;
        this->tahun_produksi = tahun_produksi;
        this->warna = warna;
    }

    //  Getter and Setter
    string getPlatNomor() {
        return plat_nomor;
    }

    void setPlatNomor(string plat_nomor) {
        this->plat_nomor = plat_nomor;
    }

    string getMerk() {
        return merk;
    }

    void setMerk(string merk) {
        this->merk = merk;
    }

    int getTahunProduksi() {
        return tahun_produksi;
    }

    void setTahunProduksi(int tahun_produksi) {
        this->tahun_produksi = tahun_produksi;
    }

    string getWarna() {
        return warna;
    }

    void setWarna(string warna) {
        this->warna = warna;
    }

    // Destructor
    ~Vehicle() {
    }
};